test/shared/TestApplicationErrorLogger.cs

```diff
@@ -44,14 +45,23 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             if (logLevel == LogLevel.Critical && ThrowOnCriticalErrors)
 #endif
             {
-                Console.WriteLine($"Log {logLevel}[{eventId}]: {formatter(state, exception)} {exception?.Message}");
+                var log = $"Log {logLevel}[{eventId}]: {formatter(state, exception)} {exception?.Message}";
+
+                Console.WriteLine(log);
 
                 if (logLevel == LogLevel.Critical && ThrowOnCriticalErrors)
                 {
-                    throw new Exception("Unexpected critical error.", exception);
+                    throw new Exception($"Unexpected critical error. {log}", exception);
                 }
             }
 
+            // Fail tests where not all the connections close during server shutdown.
+            if (eventId.Id == 21 && eventId.Name == nameof(KestrelTrace.NotAllConnectionsAborted))
```

Comment:
Ugh. Do we need both name and id here?

Reply:
It's the easiest way to not accidentally collide with a LibuvTrace.